add annotation

Dungeonmir opened the pull request.

Dungeonmir committed:
add annotation

AlekseiMaslakov commented on src/Lab2.java:
```diff
@@ -0,0 +1,74 @@
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+@interface Repeat {
```
Про вложенные классы - аналогично лаб1

AlekseiMaslakov commented on src/Lab2.java:
```diff
@@ -0,0 +1,74 @@
+    int times();
+}
+
+class Processor {
```
<img width="663" height="55" alt="image" src="https://github.com/user-attachments/assets/981d1547-2922-4463-8ddf-6eb96554e604" />
Пока все методы без параметров

AlekseiMaslakov commented on src/Lab2.java:
```diff
@@ -0,0 +1,74 @@
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
```
Так (и в 71 строчке тоже) делать не стоит. Оператор throws используется для проброса исключения на уровень выше, чтобы его можно было обработать в месте вызова метода, а не непосредственно в методе. Но выше метода main ничего нет, поэтому проброшенная ошибка так и останется никем никогда необработанной.

AlekseiMaslakov commented on src/Lab2.java:
```diff
@@ -0,0 +1,74 @@
+}
+
+class Executor {
+    public static void invokeAnnotatedMethods(Processor processor) throws Exception {
```
Желательно чётко указывать типы исключений, которые кидает метод, так читающему/использующему код более понятно с чём он имеет дело и что должен будет обработать в месте вызова.

Dungeonmir committed:
Extract classes

- extract Executor, Processor, Repeat classes.
- add default params to methods
- getArgs method to provide default params

AlekseiMaslakov commented on src/Executor.java:
```diff
@@ -0,0 +1,53 @@
+                        System.err.println("Cannot call method " + method.getName() + ": " + e.getMessage());
+                    }
+                }
+            }
```
По заданию, вызвать нужно только защищённые и приватные методы, а не все аннотированные

Dungeonmir committed:
Fix code logic according to task

- run method if modifier is protected or private, no matter if there are any annotations present
